### ISSUE TITLE
Scons fails with error "Header physfs.h is parseable, but cannot compile the test program."

### ISSUE THREAD

**Issue by gabeotisbenson** [dxx-rebirth/dxx-rebirth#352]
I'm on Arch Linux x86_64, and have all dependencies installed.  Building fails whether I clone the repo or use the package found on the AUR.

sconf.log:
```
file /home/gabe/Coding/dxx-rebirth/SConstruct,line 4027:
	Configure(confdir = .sconf_temp)
scons: dxx: version: ComputedExtraVersion('0.59.100-718-g8d80300e0a9e','## master...origin/master\n?? .sconf_temp/\n','','8d80300e0a9e8f9b567b665fe8b077372936cb4c')
scons: Configure: DXX-Archive.3: checking version of C++ compiler 'g++' ... 
scons: Configure: 'g++ (GCC) 7.2.0'

scons: Configure: DXX-Archive.3: checking path to assembler ... 
scons: Configure: 'as'

scons: Configure: DXX-Archive.3: checking version of assembler 'as' ... 
scons: Configure: 'GNU assembler (GNU Binutils) 2.29'

scons: Configure: DXX-Archive.3: checking path to linker ... 
scons: Configure: 'ld'

scons: Configure: DXX-Archive.3: checking version of linker 'ld' ... 
scons: Configure: 'GNU ld (GNU Binutils) 2.29'

scons: Configure: DXX-Archive.3:	CXX: 'g++'

scons: Configure: DXX-Archive.3:	CPPDEFINES: None

scons: Configure: DXX-Archive.3:	CPPPATH: ['common/include', 'common/main', '.']

scons: Configure: DXX-Archive.3:	CPPFLAGS: ['-Wno-sign-compare']

scons: Configure: DXX-Archive.3:	CXXFLAGS: ['-g', '-O2', '-ftabstop=4', '-Wall', '-Wshadow', '-Werror=extra', '-Werror=format=2', '-Werror=missing-braces', '-Werror=missing-include-dirs', '-Werror=uninitialized', '-Werror=undef', '-Werror=pointer-arith', '-Werror=cast-qual', '-Werror=missing-declarations', '-Werror=redundant-decls', '-Werror=vla', '-pthread', '-funsigned-char']

scons: Configure: DXX-Archive.3:	LIBS: None

scons: Configure: DXX-Archive.3:	LINKFLAGS: []

scons: Configure: DXX-Archive.3:	$CCACHE_PREFIX: None

scons: Configure: DXX-Archive.3:	$DISTCC_HOSTS: None

scons: Configure: DXX-Archive.3: checking whether C++ compiler and linker work...
.sconf_temp/conftest_0.cpp <-
  |
  |
  |/* This test is always false.  Use a non-trivial condition to
  | * discourage external text scanners from recognizing that the block
  | * is never compiled.
  | */
  |#if 1 < -1
  |0d8d46fc
  |// 'g++' => 'g++ (GCC) 7.2.0'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=as' => 'as'
  |// 'as' => 'GNU assembler (GNU Binutils) 2.29'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=ld' => 'ld'
  |// 'ld' => 'GNU ld (GNU Binutils) 2.29'
  |
  |#endif
  |
  |
  |
  |
  |#undef main	/* avoid -Dmain=SDL_main from libSDL */
  |int main(int argc,char**argv){(void)argc;(void)argv;
  |
  |
  |;}
  |
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -Werror -Wno-sign-compare -Icommon/include -Icommon/main -I. .sconf_temp/conftest_0.cpp -o .sconf_temp/conftest_0.o
g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -Werror .sconf_temp/conftest_0.o -o .sconf_temp/conftest_0
scons: Configure: yes

scons: Configure: DXX-Archive.3: checking whether C++ compiler accepts -std=gnu++14...
.sconf_temp/conftest_1.cpp <-
  |
  |
  |/* This test is always false.  Use a non-trivial condition to
  | * discourage external text scanners from recognizing that the block
  | * is never compiled.
  | */
  |#if 1 < -1
  |0d8d46fc
  |// 'g++' => 'g++ (GCC) 7.2.0'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=as' => 'as'
  |// 'as' => 'GNU assembler (GNU Binutils) 2.29'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=ld' => 'ld'
  |// 'ld' => 'GNU ld (GNU Binutils) 2.29'
  |
  |#endif
  |
  |
  |
  |
  |#undef main	/* avoid -Dmain=SDL_main from libSDL */
  |int main(int argc,char**argv){(void)argc;(void)argv;
  |
  |
  |;}
  |
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -std=gnu++14 -Werror -Wno-sign-compare -Icommon/include -Icommon/main -I. .sconf_temp/conftest_1.cpp -o .sconf_temp/conftest_1.o
scons: Configure: yes

scons: Configure: DXX-Archive.3: checking endian to use...little

scons: Configure: DXX-Archive.3: checking whether to enable word alignment fixups...no

scons: Configure: DXX-Archive.3: building with OpenGL

scons: Configure: DXX-Archive.3: checking whether to enable release options...yes

scons: Configure: DXX-Archive.3: checking whether to enable memory allocation tracking...no

scons: Configure: DXX-Archive.3: checking whether to enable level editor...no

scons: Configure: DXX-Archive.3: checking whether to enable IPv6 support...no

scons: Configure: DXX-Archive.3: checking whether to enable multiplayer over UDP...yes

scons: Configure: DXX-Archive.3: checking whether to enable UDP game tracker...yes

scons: Configure: DXX-Archive.3: checking for usable library physfs...
.sconf_temp/conftest_2.cpp <-
  |
  |
  |/* This test is always false.  Use a non-trivial condition to
  | * discourage external text scanners from recognizing that the block
  | * is never compiled.
  | */
  |#if 1 < -1
  |0d8d46fc
  |// 'g++' => 'g++ (GCC) 7.2.0'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=as' => 'as'
  |// 'as' => 'GNU assembler (GNU Binutils) 2.29'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=ld' => 'ld'
  |// 'ld' => 'GNU ld (GNU Binutils) 2.29'
  |
  |#endif
  |
  |#define DXX_WORDS_BIGENDIAN 0
  |#define DXX_WORDS_NEED_ALIGNMENT 0
  |#define DXX_USE_EDITOR 0
  |#define DXX_USE_IPv6 0
  |#define DXX_USE_UDP 1
  |#define DXX_USE_TRACKER 1
  |
  |#include <physfs.h>
  |
  |#undef main	/* avoid -Dmain=SDL_main from libSDL */
  |int main(int argc,char**argv){(void)argc;(void)argv;
  |
  |	PHYSFS_File *f;
  |	char b[1] = {0};
  |	PHYSFS_init("");
  |	f = PHYSFS_openWrite("a");
  |	PHYSFS_sint64 w = PHYSFS_write(f, b, 1, 1);
  |	(void)w;
  |	PHYSFS_close(f);
  |	f = PHYSFS_openRead("a");
  |	PHYSFS_sint64 r = PHYSFS_read(f, b, 1, 1);
  |	(void)r;
  |	PHYSFS_close(f);
  |
  |
  |;}
  |
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -std=gnu++14 -Werror -Wno-sign-compare -Icommon/include -Icommon/main -I. .sconf_temp/conftest_2.cpp -o .sconf_temp/conftest_2.o
.sconf_temp/conftest_2.cpp: In function ‘int main(int, char**)’:
.sconf_temp/conftest_2.cpp:33:43: error: ‘PHYSFS_sint64 PHYSFS_write(PHYSFS_File*, const void*, PHYSFS_uint32, PHYSFS_uint32)’ is deprecated [-Werror=deprecated-declarations]
  PHYSFS_sint64 w = PHYSFS_write(f, b, 1, 1);
                                           ^
In file included from .sconf_temp/conftest_2.cpp:24:0:
/usr/include/physfs.h:1362:27: note: declared here
 PHYSFS_DECL PHYSFS_sint64 PHYSFS_write(PHYSFS_File *handle,
                           ^~~~~~~~~~~~
.sconf_temp/conftest_2.cpp:37:42: error: ‘PHYSFS_sint64 PHYSFS_read(PHYSFS_File*, void*, PHYSFS_uint32, PHYSFS_uint32)’ is deprecated [-Werror=deprecated-declarations]
  PHYSFS_sint64 r = PHYSFS_read(f, b, 1, 1);
                                          ^
In file included from .sconf_temp/conftest_2.cpp:24:0:
/usr/include/physfs.h:1333:27: note: declared here
 PHYSFS_DECL PHYSFS_sint64 PHYSFS_read(PHYSFS_File *handle,
                           ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
scons: Configure: no

scons: Configure: DXX-Archive.3: checking for usable header physfs.h...
.sconf_temp/conftest_3.cpp <-
  |
  |
  |/* This test is always false.  Use a non-trivial condition to
  | * discourage external text scanners from recognizing that the block
  | * is never compiled.
  | */
  |#if 1 < -1
  |0d8d46fc
  |// 'g++' => 'g++ (GCC) 7.2.0'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=as' => 'as'
  |// 'as' => 'GNU assembler (GNU Binutils) 2.29'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=ld' => 'ld'
  |// 'ld' => 'GNU ld (GNU Binutils) 2.29'
  |
  |#endif
  |
  |#define DXX_WORDS_BIGENDIAN 0
  |#define DXX_WORDS_NEED_ALIGNMENT 0
  |#define DXX_USE_EDITOR 0
  |#define DXX_USE_IPv6 0
  |#define DXX_USE_UDP 1
  |#define DXX_USE_TRACKER 1
  |
  |#include <physfs.h>
  |
  |#undef main	/* avoid -Dmain=SDL_main from libSDL */
  |int main(int argc,char**argv){(void)argc;(void)argv;
  |
  |	PHYSFS_File *f;
  |	char b[1] = {0};
  |	PHYSFS_init("");
  |	f = PHYSFS_openWrite("a");
  |	PHYSFS_sint64 w = PHYSFS_write(f, b, 1, 1);
  |	(void)w;
  |	PHYSFS_close(f);
  |	f = PHYSFS_openRead("a");
  |	PHYSFS_sint64 r = PHYSFS_read(f, b, 1, 1);
  |	(void)r;
  |	PHYSFS_close(f);
  |
  |
  |;}
  |
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -std=gnu++14 -Werror -Wno-sign-compare -Icommon/include -Icommon/main -I. .sconf_temp/conftest_3.cpp -o .sconf_temp/conftest_3.o
.sconf_temp/conftest_3.cpp: In function ‘int main(int, char**)’:
.sconf_temp/conftest_3.cpp:33:43: error: ‘PHYSFS_sint64 PHYSFS_write(PHYSFS_File*, const void*, PHYSFS_uint32, PHYSFS_uint32)’ is deprecated [-Werror=deprecated-declarations]
  PHYSFS_sint64 w = PHYSFS_write(f, b, 1, 1);
                                           ^
In file included from .sconf_temp/conftest_3.cpp:24:0:
/usr/include/physfs.h:1362:27: note: declared here
 PHYSFS_DECL PHYSFS_sint64 PHYSFS_write(PHYSFS_File *handle,
                           ^~~~~~~~~~~~
.sconf_temp/conftest_3.cpp:37:42: error: ‘PHYSFS_sint64 PHYSFS_read(PHYSFS_File*, void*, PHYSFS_uint32, PHYSFS_uint32)’ is deprecated [-Werror=deprecated-declarations]
  PHYSFS_sint64 r = PHYSFS_read(f, b, 1, 1);
                                          ^
In file included from .sconf_temp/conftest_3.cpp:24:0:
/usr/include/physfs.h:1333:27: note: declared here
 PHYSFS_DECL PHYSFS_sint64 PHYSFS_read(PHYSFS_File *handle,
                           ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
scons: Configure: no

scons: Configure: DXX-Archive.3: checking for parseable header physfs.h...
.sconf_temp/conftest_4.cpp <-
  |
  |
  |/* This test is always false.  Use a non-trivial condition to
  | * discourage external text scanners from recognizing that the block
  | * is never compiled.
  | */
  |#if 1 < -1
  |0d8d46fc
  |// 'g++' => 'g++ (GCC) 7.2.0'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=as' => 'as'
  |// 'as' => 'GNU assembler (GNU Binutils) 2.29'
  |// 'g++ -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -print-prog-name=ld' => 'ld'
  |// 'ld' => 'GNU ld (GNU Binutils) 2.29'
  |
  |#endif
  |
  |#define DXX_WORDS_BIGENDIAN 0
  |#define DXX_WORDS_NEED_ALIGNMENT 0
  |#define DXX_USE_EDITOR 0
  |#define DXX_USE_IPv6 0
  |#define DXX_USE_UDP 1
  |#define DXX_USE_TRACKER 1
  |
  |#include <physfs.h>
  |
  |#undef main	/* avoid -Dmain=SDL_main from libSDL */
  |int main(int argc,char**argv){(void)argc;(void)argv;
  |
  |
  |;}
  |
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -std=gnu++14 -Werror -Wno-sign-compare -Icommon/include -Icommon/main -I. .sconf_temp/conftest_4.cpp -o .sconf_temp/conftest_4.o
scons: Configure: yes
```

**Comment by vLKp** [comment]
Core I/O functions `PHYSFS_read` / `PHYSFS_write` were marked deprecated in the PhysFS release you installed.  Unfortunately, the new non-deprecated interfaces are not present in older still widely deployed PhysFS releases (including the one I have installed), so any package that switches to the new interfaces will break for everyone on the old interfaces.  This will probably break anything using PHYSFS with `-Werror=deprecated-declarations`, `-Werror`, or a few other possibilities.

As a workaround, pass `-Wno-deprecated-declarations`, define `PHYSFS_DEPRECATED` to an empty token, or switch to a PHYFS version that does not deprecate the main interface historically used for file I/O.  Fortunately, defining `PHYSFS_DEPRECATED` has no impact on PhysFS 2.0, so it should be safe to use even on releases where it is unnecessary.

Please try this patch as an implementation of the second workaround.  It works for me (and I see it passing the desired preprocessor argument), but since my install of PhysFS does not have any deprecated functions, success for me does not guarantee it will work for you.  If it works for you, I will publish it so that others do not need any special handling.

```diff
diff --git a/SConstruct b/SConstruct
index 560a6ef1a..304e81055 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3811,6 +3811,7 @@ class DXXCommon(LazyObjectConstructor):
 			CXXFLAGS = ['-funsigned-char'],
 			CPPPATH = ['common/include', 'common/main', '.'],
 			CPPFLAGS = SCons.Util.CLVar('-Wno-sign-compare'),
+			CPPDEFINES = [('PHYSFS_DEPRECATED',)],
 		)
 		add_flags = defaultdict(list)
 		if user_settings.builddir:
```

**Comment by gabeotisbenson** [comment]
I still get the same error, here's the log:

[sconf.log](https://github.com/dxx-rebirth/dxx-rebirth/files/1404499/sconf.log)

I'm slightly out of my depth here, but I believe I applied the patch correctly.  Let me know if it's obvious that I've done something wrong.  Really appreciate the help!

**Comment by vLKp** [comment]
You applied the patch correctly.  I see in the log the new preprocessor define, exactly as expected:

```
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -std=gnu++14 -Werror -Wno-sign-compare -DPHYSFS_DEPRECATED -Icommon/include -Icommon/main -I. .sconf_temp/conftest_3.cpp -o .sconf_temp/conftest_3.o
```

Your problem now is different and strange.  It still fails for you with the overall error that Rebirth cannot use this header, but the specific test failure that leads Rebirth to that conclusion has changed.  The original failure was easy to explain: calling a deprecated function while compiling with `-Werror=deprecated-declarations`, which makes it a hard error to call a deprecated function.

Your new failure is strange:

```
g++ -c -g -O2 -ftabstop=4 -Wall -Wshadow -Werror=extra -Werror=format=2 -Werror=missing-braces -Werror=missing-include-dirs -Werror=uninitialized -Werror=undef -Werror=pointer-arith -Werror=cast-qual -Werror=missing-declarations -Werror=redundant-decls -Werror=vla -pthread -funsigned-char -std=gnu++14 -Werror -Wno-sign-compare -DPHYSFS_DEPRECATED -Icommon/include -Icommon/main -I. .sconf_temp/conftest_3.cpp -o .sconf_temp/conftest_3.o
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
<command-line>:0:19: error: expected initializer before numeric constant
.sconf_temp/conftest_3.cpp: In function ‘int main(int, char**)’:
.sconf_temp/conftest_3.cpp:33:20: error: ‘PHYSFS_write’ was not declared in this scope
  PHYSFS_sint64 w = PHYSFS_write(f, b, 1, 1);
                    ^~~~~~~~~~~~
.sconf_temp/conftest_3.cpp:33:20: note: suggested alternative: ‘PHYSFS_init’
```

I tested locally before giving you the first patch and observed no errors.  Further, if defining that macro was going to cause a problem, I would have expected it to manifest very differently.  I don't see any numeric constants on the command line that it should be complaining about.  Could you tell me the specific version of the Arch package you used to install PhysFS?  I would like to fetch the same files and test locally.  I don't use Arch, so I can't help with its package manager.  If you don't know what version you have, start with showing the output of `grep 'define PHYSFS_VER' /usr/include/physfs.h`, which will give me the specific upstream version.  That may be close enough for me to reproduce the problem.

**Comment by gabeotisbenson** [comment]
The Arch package version is 3.0.0-2

The version from /usr/include/physfs.h is 3.0.0

**Comment by vLKp** [comment]
I fetched the Arch package and temporarily switched to use its `physfs.h` header instead of my own.  The problem was not anything done by Arch, but simply that the patch I gave you was wrong in a way that only mattered when `PHYSFS_DEPRECATED` is used.  My headers don't use it, so my trivial test succeeded; yours do, so the build broke for you.

Using the Arch header reproduced the problem, and now I feel bad for not spotting this earlier.  When the preprocessor is invoked with `-DVAR`, it is equivalent to `-DVAR=1`, so the patch I told you to use defined `PHYSFS_DEPRECATED` to a literal `1` rather than the empty sequence I intended to use.  Based on where `PHYSFS_DEPRECATED` appears in `physfs.h`, expanding it to `1` would cause syntax errors.  The compiler could have reported those a bit more cleanly, but that explains the `expected initializer before numeric constant` message.

Try this instead of the patch I proposed above:

```diff
diff --git a/SConstruct b/SConstruct
index 560a6ef1a..3c049517e 100644
--- a/SConstruct
+++ b/SConstruct
@@ -3811,6 +3811,7 @@ class DXXCommon(LazyObjectConstructor):
 			CXXFLAGS = ['-funsigned-char'],
 			CPPPATH = ['common/include', 'common/main', '.'],
 			CPPFLAGS = SCons.Util.CLVar('-Wno-sign-compare'),
+			CPPDEFINES = [('PHYSFS_DEPRECATED', '')],
 		)
 		add_flags = defaultdict(list)
 		if user_settings.builddir:
```

This should define `PHYSFS_DEPRECATED` as an empty macro, inhibiting its use in `physfs.h` without causing unwanted side effects.
